examples/signal/plot_covariance_estimation.py

```diff
@@ -0,0 +1,208 @@
+"""
+===============================================================================
+Estimate covariance with different time windows
+===============================================================================
+
+Covariance estimators comparison for different EEG signal lengths and their
+impact on classification [1]_.
+"""
+# Author: Sylvain Chevallier
+#
+# License: BSD (3-clause)
+
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from matplotlib import pyplot as plt
+from mne import Epochs, pick_types, events_from_annotations
+from mne.io import concatenate_raws
+from mne.io.edf import read_raw_edf
+from mne.datasets import eegbci
+from pyriemann.estimation import Covariances
+from pyriemann.utils.distance import distance
+from pyriemann.classification import MDM
+from sklearn.model_selection import cross_val_score, StratifiedKFold
+from sklearn.pipeline import make_pipeline
+
+
+rs = np.random.RandomState(42)
+
+###############################################################################
+# Estimating covariance on synthetic data
+# ----------------------------------------
+#
+# Generate synthetic data, sampled from a distribution considered as the
+# groundtruth.
+
+n_trials, n_channels, n_times = 10, 5, 1000
+var = 2.0 + 0.1 * rs.randn(n_trials, n_channels)
+A = 2 * rs.rand(n_channels, n_channels) - 1
+A /= np.linalg.norm(A, axis=1)[:, np.newaxis]
+true_cov = np.empty(shape=(n_trials, n_channels, n_channels))
+X = np.empty(shape=(n_trials, n_channels, n_times))
+for i in range(n_trials):
+    true_cov[i] = A @ np.diag(var[i]) @ A.T
+    X[i] = rs.multivariate_normal(
+        np.array([0.0] * n_channels), true_cov[i], size=n_times
+    ).T
+
+###############################################################################
+# Covariances() object offers several estimators: SCM, Ledoit-Wolf (LWF),
+# Schaefer-Strimmer (SCH), oracle approximating shrunk covariance (OAS),
+# minimum covariance determinant (MCD) and others. We will compare the
+# distance of LWF, OAS and SCH estimators with the groundtruth, while
+# increasing epoch length.
+
+estimators = ["lwf", "oas", "sch"]
+w_len = np.linspace(10, n_times, 20, dtype=np.int)
```

Comment:
```suggestion
w_len = np.linspace(10, n_times, 20, dtype=int)
```
to remove the warning:
> DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`.